### PR TITLE
Take integrations tests out of unit tests folder

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes
It's a bit confusing where to draw the line between integration and unit, if consider any interaction with the file system as integration that needs to be taken out but that seems a bit too much since most of these tests have very limited scope and one isolated so for that only one actual test that wasn't is the `zen_server`. I also would like to challenge the idea of keeping the zenml integrations tests folder under the unit instead of having it part of the integration, would it be possible to have a unit test workflow just with zenml not with all integrations installed? 

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Comment by strickvl** [comment]
@safoinme 100% we can split up the tests that way. IMO we should have:

1) unit tests that can be run without extra integrations installed (i.e. for core zenml functionality)
2) integration tests that mainly deal with functionality that requires things to be installed (some will run in isolated environments with just 1 integration, and others will run in an environment that has a few integrations installed)
3) end-to-end tests (which run in isolated environments that run full pipelines for specific integrations + also that touch deployed cloud infra)

We'll make those changes in coming weeks, I think, but this PR is a good start towards that.

**Comment by strickvl** [comment]
@safoinme can you give an example? Also are you making sure that anything you're removing from unit tests might no longer need to be installed for unit test runners etc (so making any updates to the CI installation)? I'm not sure if that applies here.

**Review by strickvl — APPROVED** [review]
LGTM, pending tests passing ofc. Remaining pieces to this:

- handling caching (v important)
- monitoring the speed / load on our runners and how we can best allocate those resources
- Windows runners on the Azure cluster?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -136,14 +167,28 @@ jobs:
...
+      - custom-ubuntu-runners-setup-and-test-python
     strategy:
       matrix:
-        os: [macos-latest]
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
-        test_environment: ["default"]
+        os: [ubuntu-dind-runners]
+        # Disabled temporarily while CI is overhauled
+        python-version: ["3.8", "3.9", "3.10"]
+        # python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (avishniakov) [inline .github/workflows/ci.yml:178]: pls remove commented out code

**Review by avishniakov — APPROVED** [review]
Agree with @strickvl comments

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -39,11 +39,24 @@ jobs:
...
+#    strategy:
+#      matrix:
+#        os: [ubuntu-latest]
+#        python-version: ["3.8", "3.9", "3.10", "3.11"]
+#      fail-fast: false
+#    uses: ./.github/workflows/setup-python-environment.yml
+#    with:
+#      python-version: ${{ matrix.python-version }}
+#      os: ${{ matrix.os }}
+#    secrets: inherit
```
Comment (avishniakov) [inline .github/workflows/ci.yml:53]: pls remove commented out code

**Pull request merged**